File: 🟡 Medium/1894 - Find the Student that Will Replace the Chalk/_1894FindtheStudentthatWillReplacetheChalk.py
class Solution:
    def chalkReplacer(self, chalk: List[int], k: int) -> int:
        chalkSum = 0

        for i in range(len(chalk)):
            chalkSum += chalk[i]

        k %= chalkSum
        for i in range(len(chalk)):
            idx = i
            if chalk[i] > k:
                return i
            else:
                k -= chalk[i]

        return idx

File: 🟡 Medium/1894 - Find the Student that Will Replace the Chalk/_1894FindtheStudentthatWillReplacetheChalk.java
class Solution {
    public int chalkReplacer(int[] chalk, int k) {
        long chalkSum = 0;

        for (int chalkPiece : chalk) {
            chalkSum += chalkPiece;
        }

        k %= chalkSum;
        int idx;
        for (idx = 0; idx < chalk.length; idx++) {
            if (chalk[idx] > k) {
                return idx;
            } else {
                k -= chalk[idx];
            }
        }

        return idx;
    }
}